### PR TITLE
JSON server configuration launch

### PR DESCRIPTION
**Pull request by dsmith1547**
### PR Checklist

- [ ] Brief description of changes
- [ ] Links to any relevant issues
- [ ] Required status checks are passing
- [ ] User cases if changes impact the user's experience
- [ ] `@mention` a maintainer to request a review

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/default/src/DicomWebProxyDataSource/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { IWebApiDataSource } from '@ohif/core';
+import { createDicomWebApi } from '../DicomWebDataSource/index';
+
+function createDicomWebProxyApi(
```
Comment (sedghi) [inline extensions/default/src/DicomWebProxyDataSource/index.js:4]: Can you add couple of lines in the documentation of this function and also explain the schema for which the data source is being used for? what does the url look like as an exmaple etc.

Comment (sedghi) [reply extensions/default/src/DicomWebProxyDataSource/index.js:4]: For instance, it seems like it needs `servers` info to be present at the url

File: extensions/default/src/DicomWebProxyDataSource/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        const response = await fetch(url);
+        let data = await response.json();
+
+        // there seem to be a couple of variations of the case for this parameter
+        const queryStudyInstanceUIDs =
+          query.get('studyInstanceUIDs') || query.get('studyInstanceUids');
+        if (!queryStudyInstanceUIDs) {
+          console.error(`No studyInstanceUids in request for '${name}'`);
+        }
+        if (data.servers && queryStudyInstanceUIDs) {
```
Comment (sedghi) [inline extensions/default/src/DicomWebProxyDataSource/index.js:29]: Shouldn't we throw here if `data.servers` don't exist? The rest of implementation relies on it

**Review by sedghi — CHANGES_REQUESTED** [review]
Great addition, had some comments. Tagging @JamesAPetts for review as well

**Comment by sedghi** [comment]
@dsmith1547 Also forgot to mention we need something similar to the docs here https://v3-docs.ohif.org/configuration/dataSources/dicom-json for explaining the proxy and what it expects at the url etc.

File: extensions/default/src/DicomWebProxyDataSource/index.js (hunk 1 above)
Comment (dsmith1547) [reply extensions/default/src/DicomWebProxyDataSource/index.js:12]: I added a short comment to the class, but the (added) markdown documentation is probably a better way to see this

**Review by JamesAPetts — COMMENTED** [review]
This functionality looks fine. But since this is for a particular niche usecase, I don't think it be should pushed to the default OHIF extension. If it is designed to be an open source contribution, that platforms other than Kheops may use, it should have its own repo, which is an extension, which houses the datasource.

Anything that is not on the ohif roadmap: https://ohif.org/roadmap/, should ideally in an extension rather than in the core repo.

If we need to change the cornerstone extension to accept optional headers, that feels like something that _should_ go core.

**Comment by sedghi** [comment]
Looking back at this, I like this to be part of the OHIF upstream. Now that we are building the Google/Azure/AWS adapters uses this almost entirely. @dsmith1547 Can you please do that? thanks

**Review by sedghi — APPROVED** [review]
Looks great thanks

File: platform/viewer/public/config/default.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ window.config = {
     prefetch: 25,
   },
   // filterQueryParam: false,
+  // enable imageLoaderAcceptHeader configuration to force a reconfiguration of WADOImageLoader to support preflight request
```
Comment (jbocce) [inline platform/viewer/public/config/default.js:27]: Do we still want these commented lines included?

File: extensions/default/src/DicomWebProxyDataSource/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ */
+function createDicomWebProxyApi(
+  dicomWebProxyConfig,
+  UserAuthenticationService
+) {
+  const { name } = dicomWebProxyConfig;
+  let dicomWebDelegate = undefined;
+
+  const implementation = {
+    initialize: async ({ params, query, url }) => {
```
[the remark this answers is not in the document]
Comment (dsmith1547) [reply extensions/default/src/DicomWebProxyDataSource/index.js:20]: I think you're right - I likely "borrowed" this part of the impl from the DicomJSONDataSource, but it looks like it must always get it from the query param

File: platform/viewer/public/config/default.js (hunk 3 above)
Comment (dsmith1547) [reply platform/viewer/public/config/default.js:27]: nope

File: extensions/default/src/DicomWebProxyDataSource/index.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      let studyInstanceUIDs = [];
+
+      // there seem to be a couple of variations of the case for this parameter
+      const queryStudyInstanceUIDs =
+        query.get('studyInstanceUIDs') || query.get('studyInstanceUids');
+      if (!queryStudyInstanceUIDs) {
+        throw new Error(`No studyInstanceUids in request for '${name}'`);
+      }
+
+      let url = query.get('url');
```
Comment (jbocce) [inline extensions/default/src/DicomWebProxyDataSource/index.js:30]: This should be `const` I think.

Comment (dsmith1547) [reply extensions/default/src/DicomWebProxyDataSource/index.js:30]: Indeed, this no longer gets modified.

**Review by jbocce — APPROVED** [review]
Thanks for the updates!

**Pull request merged**